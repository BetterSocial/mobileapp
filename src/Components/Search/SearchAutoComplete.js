import React from 'react';
import {
  TextInput,
  View,
  Text,
  StyleSheet,
  TouchableNativeFeedback,
} from 'react-native';
import SearchIcon from '../../../assets/icons/search.svg';

const SearchAutoComplete = (props) => {
  return (
    <View style={styles.container}>
      <View style={{...styles.inputContainer, ...styles.container}}>
        <SearchIcon width={16.67} height={16.67} fill="#000" />
        <TextInput
          style={{...styles.searchInput, ...styles.textInput}}
          onChangeText={props.onChangeText}
          value={props.value}
          placeholder={props.placeholder ? props.placeholder : ''}
        />
      </View>
      {typeof props.options !== 'undefined' && props.options.length > 0 ? (
        <View style={styles.box}>
          {props.options.map((value, index) => {
            return (
              <TouchableNativeFeedback key={index} onPress={() => props.onSelect(value)}>
                <View style={styles.list}>
                  <Text style={styles.label}>{value.label}</Text>
                </View>
              </TouchableNativeFeedback>
            );
          })}
        </View>
      ) : null}
    </View>
  );
};
const styles = StyleSheet.create({
  container: {
    position: 'relative',
  },
  inputContainer: {
    height: 48,
    backgroundColor: '#FFFFFF',
    borderWidth: 1,
    borderColor: '#BDBDBD',
    borderRadius: 8,
    flexDirection: 'row',
    alignItems: 'center',
    paddingLeft: 13,
  },
  searchInput: {
    width: '90%',
    height: 40,
    fontFamily: 'Inter',
    fontStyle: 'normal',
    fontWeight: 'normal',
    fontSize: 13,
    letterSpacing: -0.28,
    color: '#BDBDBD',
    marginLeft: 5,
  },
  box: {
    minHeight: 20,
    backgroundColor: '#FFFFFF',
    borderWidth: 1,
    borderColor: '#BDBDBD',
    borderRadius: 8,
    flexDirection: 'column',
    marginBottom: 2,
    position: 'absolute',
    width: '100%',
    top: 50,
    zIndex: 2000,
    elevation: 2000
  },
  list: {
    minHeight: 35,
    backgroundColor: '#e6e6e6',
    paddingLeft: 13,
    paddingRight: 13,
    paddingTop: 2,
    paddingBottom: 2,
  },
  label: {
      textTransform: 'capitalize'
  }
});
export default SearchAutoComplete;
