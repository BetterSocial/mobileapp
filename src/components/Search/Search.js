import * as React from 'react';
import {TextInput, View, StyleSheet} from 'react-native';

import SearchIcon from '../../../assets/icons/search.svg';

const Search = (props) => {
  return (
    <View style={{...styles.inputContainer, ...styles.container}}>
      <SearchIcon width={16.67} height={16.67} fill="#000" />
      <TextInput
        style={{...styles.searchInput, ...styles.textInput}}
        onChangeText={props.onChangeText}
        value={props.value}
        placeholder={props.placeholder ? props.placeholder : ''}
      />
    </View>
  );
};
const styles = StyleSheet.create({
  inputContainer: {
    height: 48,
    backgroundColor: '#FFFFFF',
    borderWidth: 1,
    borderColor: '#BDBDBD',
    borderRadius: 8,
    flexDirection: 'row',
    alignItems: 'center',
    paddingLeft: 13,
  },
  searchInput: {
    width: '90%',
    height: 40,
    fontFamily: 'Inter',
    fontStyle: 'normal',
    fontWeight: 'normal',
    fontSize: 13,
    letterSpacing: -0.28,
    color: '#BDBDBD',
    marginLeft: 5,
  },
});
export default Search;
