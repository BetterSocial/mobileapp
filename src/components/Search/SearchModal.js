import * as React from 'react';
import Modal from 'react-native-modal';
import PropTypes from 'prop-types';
import {
    Dimensions,
    StyleSheet,
    Text,
    TouchableNativeFeedback,
    View,
} from 'react-native';

import AutoFocusTextArea from '../TextArea/AutoFocusTextArea';
import CrossIcon from '../../../assets/icons/cross.svg';
import StringConstant from '../../utils/string/StringConstant';
import { displayFormattedSearchLocations, displayFormattedSearchLocationsV2 } from '../../utils/string/StringUtils';

const {width} = Dimensions.get('screen');

const SearchModal = (props) => {
    const textRef = React.useRef(null);
    const [focus, setFocus] = React.useState(false);

    React.useEffect(() => {
        if (props.isVisible) {
        }
    }, [props]);

    console.log(`==============`)

    return (
        <Modal
            testID={'modal'}
            isVisible={props.isVisible}
            onSwipeComplete={props.onClose}
            swipeDirection={['down']}
            style={styles.view}>
            <View style={styles.container}>
                <View style={styles.space} />
                <View style={styles.content}>
                    <View style={styles.header}>
                        <TouchableNativeFeedback onPress={props.onClose}>
                            <CrossIcon width={18} height={18} fill="#000" />
                        </TouchableNativeFeedback>
                        <Text style={styles.textSearch}>
                            {StringConstant.searchModalTitle}
                        </Text>
                    </View>
                    <View style={styles.containerInput}>
                        <View style={{ ...styles.inputContainer, ...styles.containerInput }}>
                            <AutoFocusTextArea
                                keyboardAppearDelay={50}
                                autoCapitalize={'words'}
                                autoFocus={focus}
                                ref={textRef}
                                style={{ ...styles.searchInput, ...styles.textInput }}
                                onChangeText={props.onChangeText}
                                value={props.value}
                                placeholder={props.placeholder ? props.placeholder : ''}
                                placeholderTextColor="#BDBDBD"
                                multiline={false}
                                returnKeyType="search"
                            />
                        </View>
                        {props.isLoading ? <Text>Please wait...</Text> : null}

                        {typeof props.options !== 'undefined' &&
                            props.options.length > 0 ? (
                            <View style={styles.box}>
                                {props.options.map((value, index) => 
                                    // let firstLetter = value.neighborhood.split(' ');
                                    // let lastLetter = value.neighborhood.replace(
                                    //     `${firstLetter[0]} `,
                                    //     ' ',
                                    // );
                                     (
                                        <TouchableNativeFeedback
                                            key={index}
                                            onPress={() => props.onSelect(value)}>
                                            <View style={styles.list}>
                                                {/* {displayFormattedSearchLocations(props.value, value)} */}
                                                {displayFormattedSearchLocationsV2(props.value, value)}
                                            </View>
                                        </TouchableNativeFeedback>
                                    )
                                )}
                            </View>
                        ) : null}
                    </View>
                </View>
            </View>
        </Modal>
    );
};
const styles = StyleSheet.create({
    view: {
        justifyContent: 'flex-end',
        margin: 0,
    },
    container: {
        width,
        flexDirection: 'column',
        flex: 1,
    },
    space: { flex: 1 },
    content: {
        backgroundColor: 'white',
        flexDirection: 'column',
        borderColor: 'rgba(0, 0, 0, 0.1)',
        borderTopLeftRadius: 16,
        borderTopRightRadius: 16,
        flex: 8,
        width,
        padding: 22,
    },
    contentTitle: {
        fontSize: 20,
        marginBottom: 12,
    },
    header: {
        flexDirection: 'row',
        alignItems: 'center',
    },
    textSearch: {
        // fontFamily: 'SFProText-Regular',
        fontStyle: 'normal',
        fontWeight: 'bold',
        fontSize: 18,
        textAlign: 'center',
        letterSpacing: -0.28,
        color: '#000000',
        paddingLeft: 19,
    },
    inputContainer: {
        height: 48,
        width: width - 44,
        backgroundColor: '#FFFFFF',
        borderWidth: 1,
        borderColor: '#BDBDBD',
        borderRadius: 8,
        flexDirection: 'row',
        alignItems: 'center',
        paddingLeft: 13,
        marginTop: 26,
    },
    containerInput: {
        position: 'relative',
    },
    searchInput: {
        width: '90%',
        height: 40,
        fontFamily: 'Inter',
        fontStyle: 'normal',
        fontWeight: 'normal',
        fontSize: 14,
        letterSpacing: -0.28,
        color: '#000000',
        marginLeft: 5,
    },
    box: {
        minHeight: 20,
        backgroundColor: '#FFFFFF',
        borderRadius: 8,
        flexDirection: 'column',
        marginBottom: 2,
        position: 'absolute',
        width: '100%',
        top: 90,
        zIndex: 2000,
        elevation: 2000,
    },
    list: {
        minHeight: 35,
        paddingTop: 2,
        paddingBottom: 2,
        flexDirection: 'row',
        alignItems: 'center',
    },
    labelBold: {
        fontFamily: 'Inter',
        fontStyle: 'normal',
        fontWeight: 'bold',
        fontSize: 14,
        lineHeight: 17,
        color: '#000000',
        textTransform: 'capitalize',
    },
    label: {
        fontFamily: 'Inter',
        fontStyle: 'normal',
        fontWeight: '600',
        fontSize: 14,
        lineHeight: 17,
        color: '#000000',
        textTransform: 'capitalize',
    },
});

SearchModal.prototypes = {
    isVisible: PropTypes.bool,
    isLoading: PropTypes.bool,
    options: PropTypes.array,
}

export default SearchModal;
