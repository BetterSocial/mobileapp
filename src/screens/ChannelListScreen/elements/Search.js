/* eslint-disable no-use-before-define */
/* eslint-disable no-underscore-dangle */
import * as React from 'react';
import PropTypes from 'prop-types';
import {Pressable, StyleSheet, Text, TouchableOpacity, View} from 'react-native';
import {useNavigation} from '@react-navigation/core';

import MemoIcNewChat from '../../../assets/icons/ic_new_chat';
import MemoIcSearch from '../../../assets/icons/Ic_search';
import StringConstant from '../../../utils/string/StringConstant';
import dimen from '../../../utils/dimen';
import {DISCOVERY_TAB_USERS} from '../../../utils/constants';
import {SIZES} from '../../../utils/theme';
import {colors} from '../../../utils/colors';
import {fonts} from '../../../utils/fonts';

const Search = ({route, onPress, isShowNewChat = true}) => {
  const navigation = useNavigation();

  const __handleOnSearchClicked = () => {
    navigation.push('DiscoveryScreen', {
      tab: DISCOVERY_TAB_USERS
    });
  };

  const themeColor = () => {
    if (route?.name === 'SignedChannelList') return colors.darkBlue;
    return colors.anon_primary;
  };

  return (
    <View style={styles.animatedViewContainer}>
      <Pressable onPress={__handleOnSearchClicked} style={styles.searchPressableContainer}>
        <View style={styles.wrapperSearch}>
          <View style={styles.wrapperIcon}>
            <MemoIcSearch width={16.67} height={16.67} />
          </View>
          <Text style={styles.input}>{StringConstant.chatTabHeaderPlaceholder}</Text>
        </View>
      </Pressable>
      {isShowNewChat && (
        <TouchableOpacity style={styles.wrapperButton} onPress={onPress}>
          <Text style={[styles.newPostText, {color: themeColor()}]}>
            {StringConstant.chatTabHeaderCreateChatButtonText}
          </Text>
          <View>
            <MemoIcNewChat height={18} width={16} color={themeColor()} style={styles.newChatIcon} />
          </View>
        </TouchableOpacity>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    backgroundColor: 'white',
    marginBottom: SIZES.base
  },
  searchPressableContainer: {
    flex: 1
  },
  wrapperSearch: {
    flex: 1,
    flexDirection: 'row',
    backgroundColor: '#F5F5F5',
    marginLeft: 20,
    marginRight: 12,
    borderRadius: 8,
    alignSelf: 'center',
    height: 36
  },
  wrapperButton: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around',
    marginEnd: SIZES.base,
    paddingRight: 12,
    paddingTop: 9,
    paddingBottom: 9
  },
  input: {
    paddingStart: 10,
    lineHeight: 36,
    flex: 1,
    paddingBottom: 0,
    paddingTop: 0,
    fontFamily: fonts.inter[400],
    fontSize: 14,
    alignSelf: 'center',
    color: colors.gray1
  },
  wrapperIcon: {
    marginLeft: 8,
    alignSelf: 'center',
    justifyContent: 'center'
  },
  newPostText: {
    marginRight: dimen.normalizeDimen(5),
    fontFamily: 'Inter-SemiBold',
    fontSize: 12,
    lineHeight: 14.52
  },
  newChatIcon: {
    marginTop: 0
  },
  animatedViewContainer: {
    flexDirection: 'row',
    backgroundColor: 'white',
    marginBottom: SIZES.base,
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    zIndex: 10,
    paddingTop: 7,
    paddingBottom: 7,
    borderBottomWidth: 1,
    borderBottomColor: colors.alto
  }
});

Search.propTypes = {
  route: PropTypes.object.isRequired,
  onPress: PropTypes.func.isRequired
};

export default Search;
