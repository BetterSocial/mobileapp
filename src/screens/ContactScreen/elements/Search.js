import * as React from 'react';
import {
  View,
  Text,
  TouchableOpacity,
  StyleSheet,
  TextInput,
  Animated,
} from 'react-native';

import MemoIcNewChat from '../../../assets/icons/ic_new_chat';
import MemoIc_search from '../../../assets/icons/Ic_search';
import StringConstant from '../../../utils/string/StringConstant';
import {COLORS, FONTS, SIZES} from '../../../utils/theme';

const Search = ({onPress, animatedValue}) => {
  return (
    <Animated.View style={styles.animatedViewContainer(animatedValue)}>
      <View style={styles.wrapperSearch}>
        <TextInput
          multiline={false}
          placeholder={'Search Users'}
          style={styles.input}
        />
        <View style={styles.wrapperIcon}>
          <MemoIc_search width={20} height={20} />
        </View>
      </View>
    </Animated.View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    backgroundColor: 'white',
    marginBottom: SIZES.base,
  },
  wrapperSearch: {
    flex: 1,
    backgroundColor: '#F5F5F5',
    marginLeft: 15,
    marginRight: 15,
    borderRadius: SIZES.radius,
    alignSelf: 'center',
  },
  wrapperButton: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around',
    marginEnd: SIZES.base,
    paddingLeft: 8,
    paddingRight: 8,
    paddingTop: 9,
    paddingBottom: 9,
    // backgroundColor: 'red',
  },
  input: {
    marginLeft: 24,
    paddingStart: 16,
    lineHeight: 24,
  },
  wrapperIcon: {
    position: 'absolute',
    left: 8,
    top: 0,
    bottom: 0,
    justifyContent: 'center',
  },
  newPostText: {
    color: COLORS.holyTosca,
    marginRight: 11,
    ...FONTS.h4,
  },
  animatedViewContainer: (animatedValue) => ({
    flexDirection: 'row',
    backgroundColor: 'white',
    marginBottom: SIZES.base,
    marginTop: animatedValue,
    top: 0,
    left: 0,
    right: 0,
    zIndex: 10,
    padding: 7,
    borderBottomWidth: 1,
    borderBottomColor: COLORS.gray1,
  }),
});

export default Search;
