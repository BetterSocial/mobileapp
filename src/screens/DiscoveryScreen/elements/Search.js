import * as React from 'react';
import {
  Animated,
  Pressable,
  StyleSheet,
  Text,
  TextInput,
  TouchableOpacity,
  View
} from 'react-native';
import { useNavigation } from '@react-navigation/core'

import DiscoveryAction from '../../../context/actions/discoveryAction';
import DiscoveryRepo from '../../../service/discovery';
import GeneralComponentAction from '../../../context/actions/generalComponentAction';
import IconClear from '../../../assets/icon/IconClear';
import MemoIc_arrow_back_white from '../../../assets/arrow/Ic_arrow_back_white';
import MemoIc_pencil from '../../../assets/icons/Ic_pencil';
import MemoIc_search from '../../../assets/icons/Ic_search';
import StringConstant from '../../../utils/string/StringConstant';
import dimen from '../../../utils/dimen';
import {COLORS, FONTS, SIZES} from '../../../utils/theme';
import {Context} from '../../../context/Store';
import { colors } from '../../../utils/colors';
import {fonts} from '../../../utils/fonts';

let getDataTimeoutId;

const DiscoverySearch = ({onPress, animatedValue, showBackButton = false, onContainerClicked = () => {}}) => {
  const navigation = useNavigation()
  const [generalComponent, generalComponentDispatch] = React.useContext(Context).generalComponent
  const [discovery, discoveryDispatch] = React.useContext(Context).discovery
  const discoverySearchBarRef = React.useRef(null)
  const [isTextAvailable, setIsTextAvailable] = React.useState(false)
  const [isFocus, setIsFocus] = React.useState(false)

  let { discoverySearchBarText } = generalComponent

  const __handleBackPress = () => {
    if(navigation.canGoBack()) navigation.goBack()
  }

  const __handleFocus = (isFocus) => {
    setIsFocus(isFocus)
  }

  const __handleChangeText = (text) => {
    setIsTextAvailable(text.length > 0)
    GeneralComponentAction.setDiscoverySearchBar(text, generalComponentDispatch)
  }

  const __handleOnClearText = () => {
    __handleChangeText("")
    setIsTextAvailable(false)
    discoverySearchBarRef.current.focus()
  }

  const __fetchDiscoveryData = async() => {
    let data = await DiscoveryRepo.fetchDiscoveryData(discoverySearchBarText)
    if(data.success) {
      DiscoveryAction.setDiscoveryData(data, discoveryDispatch)
    }
    DiscoveryAction.setDiscoveryLoadingData(false, discoveryDispatch)
  }

  React.useEffect(() => {
    if(discoverySearchBarText.length > 1) {
      DiscoveryAction.setDiscoveryLoadingData(true, discoveryDispatch)
      if(getDataTimeoutId) clearTimeout(getDataTimeoutId)
      
      getDataTimeoutId = setTimeout( async() => {
        await __fetchDiscoveryData()
      }, 3000)
    }
  }, [discoverySearchBarText])

  React.useEffect(() => {
    const unsubscribe = () => {
      setIsTextAvailable(false)
      GeneralComponentAction.setDiscoverySearchBar('', generalComponentDispatch)
      DiscoveryAction.setDiscoveryData({
        followedUsers: [], 
        unfollowedUsers: [], 
        followedDomains: [],
        unfollowedDomains: [], 
        followedTopic: [], 
        unfollowedTopic: [], 
        news: [],
      }, discoveryDispatch)
    }

    return unsubscribe
  },[])

  return (
    <Animated.View style={styles.animatedViewContainer(animatedValue)}>
      <Pressable onPress={__handleBackPress}
        android_ripple={{
          color: COLORS.gray1,
          borderless: true,
          radius: 20,
        }} style={styles.arrowContainer}>
        <View style={styles.backArrow}>        
          <MemoIc_arrow_back_white width={20} height={12} fill={colors.black} style={{ alignSelf: 'center'}}/>
        </View>
      </Pressable>
      <Pressable onPress={onContainerClicked} style={styles.searchContainer}>
        <View style={styles.wrapperSearch}>
          {(!isTextAvailable && !isFocus) && <View style={styles.wrapperIcon}>
            <MemoIc_search width={16.67} height={16.67} />
          </View>
          }
          {/* <Text style={styles.inputText}>{StringConstant.newsTabHeaderPlaceholder}</Text> */}
          <TextInput
            ref={discoverySearchBarRef}
            focusable={true}
            autoFocus={true}
            value={generalComponent.discoverySearchBarText}
            onChangeText={__handleChangeText}
            onFocus={() => __handleFocus(true)}
            onBlur={() => __handleFocus(false)}
            multiline={false}
            placeholder={StringConstant.discoverySearchBarPlaceholder}
            placeholderTextColor={COLORS.gray1}
            style={styles.input} />

          {(isTextAvailable || isFocus) && <Pressable onPress={__handleOnClearText} style={styles.clearIconContainer}
            android_ripple={{
              color: COLORS.gray1,
              borderless: true,
              radius: 14,
            }}>
            <View style={styles.wrapperDeleteIcon}>
              <IconClear width={9} height={10} iconColor={colors.black}/>
            </View>
          </Pressable> }
        </View>
      </Pressable>
    </Animated.View>
  );
};

const styles = StyleSheet.create({
  arrowContainer: {paddingLeft:20},
  backArrow : {flex: 1, justifyContent: 'center', marginRight: 9, },
  container: {
    flexDirection: 'row',
    backgroundColor: 'white',
    marginBottom: SIZES.base,
    marginHorizontal: SIZES.base,
  },
  clearIconContainer: {
    justifyContent: 'center', 
    alignItems: 'center',
  },
  searchContainer: {
    flex: 1,
    marginRight: 20,
  },
  wrapperSearch: {
    flex: 1,
    backgroundColor: '#F5F5F5',
    borderRadius: 8,
    alignSelf: 'center',
    flexDirection: 'row',
    height: 36,
    paddingRight: 8,
  },
  wrapperButton: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around',
    marginEnd: SIZES.base,
    paddingLeft: 8,
    paddingRight: 8,
    paddingTop: 9,
    paddingBottom: 9,
  },
  input: {
    marginRight: 16,
    paddingStart: 8,
    flex: 1,
    fontSize: 14,
    fontFamily: fonts.inter[400],
    height: 36,
    paddingTop: 0,
    paddingBottom: 0,
  },
  inputText: {
    marginRight: 16,
    paddingStart: 10,
    flex: 1,
    fontSize: 14,
    fontFamily: fonts.inter[400],
    // height: 36,
    paddingTop: 0,
    paddingBottom: 0,
    color: COLORS.gray1,
    alignSelf: 'center'
  },
  wrapperIcon: {
    marginLeft: 9.67,
    marginRight: 1.67,
    alignSelf: 'center',
    justifyContent: 'center',
  },
  wrapperDeleteIcon: {
    alignSelf: 'center',
    justifyContent: 'center',
    marginRight: 9.5,
  },
  newPostText: {
    color: COLORS.holyTosca,
    marginRight: 11,
    ...FONTS.h3,
  },
  animatedViewContainer: (animatedValue) => ({
    flexDirection: 'row',
    backgroundColor: 'white',
    marginTop: animatedValue,
    zIndex: 10,
    height: dimen.size.DISCOVERY_HEADER_HEIGHT,
    paddingTop: 7,
    paddingBottom: 7,
    borderBottomWidth: 1,
    borderBottomColor: COLORS.alto,
    // paddingLeft: 20,
    // paddingRight: 20,
  }),
});

export default DiscoverySearch;
