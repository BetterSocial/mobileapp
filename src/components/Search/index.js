import Search from './Search';
import SearchAutoComplete from './SearchAutoComplete';
import SearchModal from './SearchModal';

export {Search, SearchAutoComplete, SearchModal};
