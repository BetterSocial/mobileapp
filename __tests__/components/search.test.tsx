import React from 'react';
import {shallow, configure} from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import Search from '../../src/components/Search/Search';
import SearchModal from '../../src/components/Search/SearchModal';
import SearchAutoComplete from '../../src/components/Search/SearchAutoComplete';

configure({adapter: new Adapter()});
describe('component Search', () => {
  it('Search snapshot', () => {
    const component = shallow(
      <Search
        onChangeText={() => {}}
        value={'Search'}
        placeholder="input search"
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it('Search renders correctly', () => {
    const tree = renderer
      .create(
        <Search
          onChangeText={() => {}}
          value={'Search'}
          placeholder="input search"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('SearchModal snapshot', () => {
    const component = shallow(
      <SearchModal
        isVisible={false}
        onClose={() => {}}
        onChangeText={() => {}}
        value="SearchModal"
        placeholder="Search"
        isLoading={false}
        options={{
          neighborhood: 'indonesia',
        }}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it('SearchModal renders correctly', () => {
    const tree = renderer
      .create(
        <SearchModal
          isVisible={false}
          onClose={() => {}}
          onChangeText={() => {}}
          value="SearchModal"
          placeholder="Search"
          isLoading={false}
          options={{
            neighborhood: 'indonesia',
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('SearchAutoComplete snapshot', () => {
    const component = shallow(
      <SearchAutoComplete
        isVisible={false}
        onClose={() => {}}
        onChangeText={() => {}}
        value="SearchModal"
        placeholder="Search"
        isLoading={false}
        options={{
          neighborhood: 'indonesia',
        }}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it('SearchAutoComplete renders correctly', () => {
    const tree = renderer
      .create(
        <SearchAutoComplete
          isVisible={false}
          onClose={() => {}}
          onChangeText={() => {}}
          value="SearchModal"
          placeholder="Search"
          isLoading={false}
          options={{
            neighborhood: 'indonesia',
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
