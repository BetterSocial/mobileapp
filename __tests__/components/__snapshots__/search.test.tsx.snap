// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component Search SearchAutoComplete snapshot 1`] = `
<View
  style={
    Object {
      "position": "relative",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#BDBDBD",
        "borderRadius": 8,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 48,
        "paddingLeft": 13,
        "position": "relative",
      }
    }
  >
    <[object Object]
      fill="#000"
      height={16.67}
      width={16.67}
    />
    <Component
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Search"
      rejectResponderTermination={true}
      style={
        Object {
          "color": "#BDBDBD",
          "fontFamily": "Inter",
          "fontSize": 13,
          "fontStyle": "normal",
          "fontWeight": "normal",
          "height": 40,
          "letterSpacing": -0.28,
          "marginLeft": 5,
          "width": "90%",
        }
      }
      underlineColorAndroid="transparent"
      value="SearchModal"
    />
  </View>
</View>
`;
